src/MorsePlayer.js

```diff
@@ -132,6 +134,19 @@ class MorsePlayer {
     return t;
   };
 
+  resetRandomFrequency = () => {
+    let frequency = -1;
+    while (
+      !(
+        frequency >= this.store.lowerBoundFrequency &&
+        frequency <= this.store.upperBoundFrequency
+      )
+    ) {
+      frequency = Math.floor(Math.random() * this.store.upperBoundFrequency);
```

Comment:
If unlucky, this can end up looping a few times. I would suggest instead something like:

frequency = Math.floor(Math.random() * (this.store.upperBoundFrequency - this.store.lowerBoundFrequency)) + this.store.lowerBoundFrequency;

then the loop wouldn't be needed at all unless I'm missing something

Reply:
I'm not sure if saving a few microseconds is significant, but your approach is cleaner. I'll go ahead and change it. It looks sane to me. :)